#20 login workflow

djkowa opened the pull request:
#20

Commit:
space

djkowa commented:
progress for the day, please review

goranesine committed:
added typedef and loginO

Commit:
welcome page created and added in routes

Commit:
created account form page structure

matejthetree commented on lib/mvos/ui/page/create_account_page.dart:
```diff
@@ -0,0 +1,21 @@
+          onPressed: () => Navigator.pop(context, RouteName.welcomePage),
+        ),
+      ),
+      body: CAForm(),
```
full name CreateAccountForm

djkowa replied:
done

matejthetree commented on lib/mvos/ui/widgets/create_account_form.dart:
```diff
@@ -0,0 +1,93 @@
+              SizedBox(height: 24,),
+              TextFormField(
+                decoration: InputDecoration(icon: Icon(Icons.person),
+                  labelText: 'Ime',
```
engleski jezik

matejthetree commented on lib/mvos/ui/widgets/create_account_form.dart:
```diff
@@ -0,0 +1,93 @@
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.all(8.0),
+      child: Form(
```
needs to be statefull widget, and you need to add form key to be able to validate the whole form

matejthetree commented on lib/mvos/ui/widgets/create_account_form.dart:
```diff
@@ -0,0 +1,93 @@
+              ),
+              RaisedButton(
+                color: Colors.pink,
+                onPressed: ()=> Navigator.pushNamed(context, RouteName.homePage),
```
this will be block function instead of arrow, because you will first validate form before sending the data trough observable. only if observable future gets resolved, you will push new route

Commit:
added full name CreateAccountForm,
added english words:name,surname,telephone
CreateAccountForm is Stateful widget now

Commit:
work in progress

Commit:
added validators to all form fields

matejthetree commented on lib/mvos/ui/page/create_account_page.dart:
```diff
@@ -0,0 +1,21 @@
+    return Scaffold(
+      appBar: AppBar(
+        automaticallyImplyLeading: false,
+        backgroundColor: Colors.pink,
```
ovo bi trebalo ici automatski iz teme

matejthetree commented on lib/mvos/ui/page/logIn_page.dart:
```diff
@@ -1,16 +1,53 @@
+            SizedBox(
+              height: 200,
+            ),
+            Form(
```
i ovo ste mogli u zaseban widget. LoginForm

matejthetree commented on lib/mvos/ui/page/welcome_page.dart:
```diff
@@ -0,0 +1,54 @@
+                color: Colors.pink,
+                child: Text(
+                  'Create Account',
+                  style: TextStyle(color: Colors.white, fontSize: 20),
```
bolje da kao textStyle koristite stil iz teme. `Theme.of(contex).textStyle.darkDisplay1` <- tako nesto

matejthetree replied:
pogledajte koje imate od opcija pa odaberite theme koji vam pase. tako ste onda unutar material guidelinesa

matejthetree commented on lib/mvos/ui/widgets/create_account_form.dart:
```diff
@@ -0,0 +1,255 @@
+                  onEditingComplete: () => _surnameFocusNode.requestFocus(),
+                  validator: (value) {
+                    if (value.length < 3) {
+                      return 'Please enter more text';
```
`Name has to have at least 3 characters`

matejthetree replied:
2

matejthetree commented on lib/mvos/ui/widgets/create_account_form.dart:
```diff
@@ -0,0 +1,255 @@
+                  controller: _surnameController,
+                  validator: (value) {
+                    if (value.length < 3) {
+                      return 'Please enter more text';
```
Surname has to have at least 2 characters

matejthetree commented on lib/mvos/ui/widgets/create_account_form.dart:
```diff
@@ -0,0 +1,255 @@
+                  controller: _phoneController,
+                  validator: (value) {
+                    if (value.length<9) {
+                      return 'Please enter more number';
```
Phone number has minimum of 9 characters

matejthetree commented on lib/mvos/ui/widgets/create_account_form.dart:
```diff
@@ -0,0 +1,255 @@
+                  controller: _passwordController,
+                  validator: (value) {
+                   if (value.length < 6) {
+                     return 'Please enter more text';
```
Minimum of 6 characters

matejthetree commented on lib/mvos/ui/widgets/create_account_form.dart:
```diff
@@ -0,0 +1,255 @@
+                  validator: (value) {
+                  if (value != _passwordController.text) {
+                    return 'Password is not matching';
+                  }return null;
```
format

matejthetree commented on lib/mvos/ui/widgets/create_account_form.dart:
```diff
@@ -0,0 +1,255 @@
+                    width: 180,
+                    height: 50,
+                    child: RaisedButton(
+                      color: Colors.pink,
```
put the color again inside theme. If it is primary color, then pink will be automatically for raised button

matejthetree reviewed (changes requested):
ok je, neke sitnice

Commit:
added themes and made changes as requested

Commit:
create account button finished

Commit:
added circular progress indicator on create account page

Commit:
login form finished

Commit:
login form finished

Commit:
matje fixes
